packages/shared/src/components/widgets/ReferralWidget.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { ReactElement } from 'react';
+import { Button, ButtonSize } from '../buttons/Button';
+import { TextField } from '../fields/TextField';
+import { link } from '../../lib/links';
+import { useShareOrCopyLink } from '../../hooks/useShareOrCopyLink';
+import { labels } from '../../lib';
+import { AnalyticsEvent, TargetId } from '../../lib/analytics';
+import ReferralSocialShareButtons from './ReferralSocialShareButtons';
+
+const ReferralWidget = ({ url }: { url: string }): ReactElement => {
+  const inviteLink = url || link.referral.defaultUrl;
+  const [, onShareOrCopyLink] = useShareOrCopyLink({
+    text: labels.referral.generic.inviteText,
+    link: inviteLink,
+    trackObject: () => ({
+      event_name: AnalyticsEvent.CopyReferralLink,
+      target_id: TargetId.ProfilePage,
+    }),
+  });
+  const onShareClick = () => {
+    onShareOrCopyLink();
+  };
```

Comment:
No need to wrap this, we can just do an inline function.
```suggestion
  const onShareClick = () => {
    onShareOrCopyLink();
  };
```